ibis-server/app/query_cache/__init__.py

```diff
@@ -70,13 +72,18 @@ def get_cache_file_timestamp(self, data_source: str, sql: str, info) -> int | No
         return None
 
     def _generate_cache_key(self, data_source: str, sql: str, info) -> str:
-        key_parts = [
-            data_source,
-            sql,
-            info.host.get_secret_value(),
-            info.port.get_secret_value(),
-            info.user.get_secret_value(),
-        ]
+        key_parts = []
+        if isinstance(info, ConnectionUrl):
+            key_parts = [data_source, sql, info.connection_url.get_secret_value()]
+        else:
+            key_parts = [
+                data_source,
+                sql,
+                info.host.get_secret_value(),
+                info.port.get_secret_value(),
+                info.user.get_secret_value(),
+            ]
+        logger.debug(f"Hash key components: {key_parts}")
```

Comment:
```suggestion
        logger.debug("Hash key components: {}", key_parts)
```
It's better to input `key_parts` as an argument.